### PR TITLE
chore: use requirements.txt in readme and staging jobs

### PR DESCRIPTION
**Pull request by mpeddada1**
Considered combining the requirements.txt files in .kokoro and .github into one but decided against it as it would result in unrelated dependencies being downloaded.

### CODE REVIEW

**Comment by mpeddada1** [comment]
Manual run of `readme` github actions job with these changes succeeded: https://github.com/googleapis/google-cloud-java/actions/runs/3251962656/jobs/5337636798

**Review by meltsufin — APPROVED** [review]
LGTM, but let's not merge until after today's release.

**Review by meltsufin — APPROVED** [review]

**Pull request merged**
